### PR TITLE
Update image improvements

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -478,9 +490,8 @@ update_git_repo() {
 	GHBRANCH="$2"
 	REPO="$3"
 	echo "`date`: Updating git repo [${NAME}] (${LOG_DIR}/git-checkout.log)"
-	(cd ${SOURCES}/${NAME} && git reset --hard) >${LOG_DIR}/git-checkout.log 2>&1 || exit_err "Failed git reset"
-	(cd ${SOURCES}/${NAME} && git fetch --unshallow) >${LOG_DIR}/git-checkout.log 2>&1
-	(cd ${SOURCES}/${NAME} && git pull origin ${GHBRANCH}) >${LOG_DIR}/git-checkout.log 2>&1 || exit_err "Failed git pull"
+	(cd ${SOURCES}/${NAME} && git fetch origin ${GHBRANCH}) >${LOG_DIR}/git-checkout.log 2>&1 || exit_err "Failed git fetch"
```
Comment (kmoore134) [inline scripts/build.sh:493]: Does this succeed with the initial repos being checked out shallow? When I tested originally I couldn't get it to work again without running --unshallow first...

File: truenas_install/__main__.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import logging
+import subprocess
+import sys
+
+logger = logging.getLogger(__name__)
+
+if __name__ == "__main__":
+    update = json.loads(sys.stdin.read())
+
+    root = update["root"]
```
Comment (william-gr) [inline truenas_install/__main__.py:13]: I wonder if we shouldn't create the new ZFS dataset from here so its unified between install and update?

File: truenas_install/__main__.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    undo = []
+    try:
+        subprocess.run(["mount", "-t", "proc", "none", f"{root}/proc"], check=True)
+        undo.append(["umount", f"{root}/proc"])
+
+        subprocess.run(["mount", "-t", "sysfs", "none", f"{root}/sys"], check=True)
+        undo.append(["umount", f"{root}/sys"])
+
+        for device in glob.glob("/dev/sd?*") + ["/dev/zfs"]:
```
Comment (william-gr) [inline truenas_install/__main__.py:34]: It could also be vd*

Wonder if there are others as well

File: scripts/build_manifest.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        for file in files:
+            abspath = os.path.join(root, file)
+            checksums[os.path.relpath(abspath, output)] = subprocess.run(
+                ["sha1sum", abspath],
+                check=True, stdout=subprocess.PIPE, encoding="utf-8", errors="ignore",
+            ).stdout.split()[0]
+
+    with open(os.path.join(output, "manifest.json"), "w") as f:
+        f.write(json.dumps({
+            "version": version,
```
Comment (william-gr) [inline scripts/build_manifest.py:28]: We lost date key? On purpose?

File: truenas_install/__main__.py (hunk 3 above)
Comment (kmoore134) [reply truenas_install/__main__.py:34]: Here's list we currently have: 
https://github.com/truenas/truenas-installer/blob/master/usr/sbin/truenas-install#L139

File: scripts/build_manifest.py (hunk 4 above)
Comment (themylogin) [reply scripts/build_manifest.py:30]: Done

File: scripts/build.sh (hunk 1 above)
Comment (themylogin) [reply scripts/build.sh:494]: Fixed it

**Comment by william-gr** [comment]
Good job!

**Pull request merged**